spec/models/ability_spec.rb

```diff
@@ -95,6 +95,67 @@
+      let(:conference_with_open_registration_and_limit_excedeed) { create(:conference, registration_limit: '1') }
+      let(:registration_for_user_with_open_registration_and_limit_excedeed) { create(:registration, conference: conference_with_open_registration_and_limit_excedeed, user: user2) }
+
+      it{ should_not be_able_to(:new, Registration.new(conference_id: registration_for_user_with_open_registration_and_limit_excedeed.conference_id)) }
+      it{ should_not be_able_to(:create, Registration.new(conference_id: registration_for_user_with_open_registration_and_limit_excedeed.conference_id)) }
+      it{ should_not be_able_to(:new, Registration.new(conference_id: user_registration.conference_id)) }
+      it{ should_not be_able_to(:create, Registration.new(conference_id: user_registration.conference_id)) }
+
+      context 'when user is a speaker' do
+        context 'when event is confirmed and the registration period is not set' do
+          let(:conference_with_speaker_confirmed) { create(:conference) }
+          let(:event_with_speaker_confirmed) { create(:event, state: 'confirmed', program: conference_with_speaker_confirmed.program) }
+
+          it 'checks that speaker with event confirmed with registration period not set cannot open new registration' do
+            event_with_speaker_confirmed.speakers << user
+            should_not be_able_to(:new, Registration.new(conference: conference_with_speaker_confirmed))
+          end
+
+          it 'checks that speaker with event confirmed with registration period not set cannot create registration' do
+            event_with_speaker_confirmed.speakers << user
+            should_not be_able_to(:create, Registration.new(conference: conference_with_speaker_confirmed))
+          end
+        end
+
+        context 'when event is confirmed and the registration period is set' do
+          let(:conference_with_speaker_confirmed) { create(:conference) }
+          let(:event_with_speaker_confirmed) { create(:event, state: 'confirmed', program: conference_with_speaker_confirmed.program) }
+
+          it 'checks that speaker with event confirmed with registration period set can open new registration' do
+            event_with_speaker_confirmed.speakers << user
+            conference_with_speaker_confirmed.registration_period = create(:registration_period)
+            should be_able_to(:new, Registration.new(conference: conference_with_speaker_confirmed))
+          end
+
+          it 'checks that speaker with event confirmed with registration period set can create registration' do
+            event_with_speaker_confirmed.speakers << user
+            conference_with_speaker_confirmed.registration_period = create(:registration_period)
+            should be_able_to(:create, Registration.new(conference: conference_with_speaker_confirmed))
+          end
+        end
+
+        context 'when registration are closed and registration_limit_exceeded' do
```

Comment:
This is not one single test case. I would expect that we test at least for the following:
- registration open + limit_exceeded 
- registration closed + limit **not** exceeded

Reply:
@differentreality Will correct it 👍